### DISCUSSION TITLE
twilio config

### DISCUSSION THREAD

**Discussion by feadrick** [Resgrid/Core#40] [Q&A] [answered]
i wanted to use twilio on the the resgrid,what  value i shuld assign to 

1. TwilioApiUrl
2. TwilioVoiceCallApiTurl
3. TwilioVoiceApiUrl

**Comment by ucswift** [comment, marked as answer]
Hey @feadrick thanks for submitting this question!

Resgrid's web systems are split into 3:
Resgrid Web is the web application with the UI that users interact with
Resgrid API is an web api project that is used for system to system communications
Resgrid Events is a web project that is used for SignalR

Those Twilio API Urls are used to send to Twilio for Twilio to call back into your Resgrid installation via the Resgrid API urls.

TwilioApiUrl handles text message iteration and will be [your resgrid api url]/api/Twilio/IncomingMessage
TwilioVoiceCallApiTurl is for outbound voice calls and will be [your resgrid api url]/api/Twilio/VoiceCall?userId={0}&callId={1}
TwilioVoiceApiUrl is for inbound voice calls and will be [your resgrid api url]/api/Twilio/InboundVoice

So if you've installed Resgrid at mycompany.com and have the API accessible via the resgridapi sub-domain your url will be:

https://resgridapi.mycompany.com/api/Twilio/IncomingMessage

for other Twilio related configs you also need to set SmsProviderType and BackupSmsProviderType to "0" in SystemBehaviorConfig to use Twilio for SMS messaging. Then in NumberProviderConfig TwilioAccountSid and TwilioAuthToken both come from your Twilio account. Finally TwilioResgridNumber is the number you've provisioned in Twilio.
